### DISCUSSION TITLE
Volume flashing support

### DISCUSSION THREAD

**Discussion by davedesro** [intercreate/smpmgr#16] [Ideas]
Today, when using `smpmgr --port /dev/ttyACM0 upgrade filename.bin`, if `/dev/ttyACM0` is not available, `smpmgr` errors out. This is a common flow for many applications

The application `tio` does things differently. If `/dev/ttyACM0` does not exist, it simply waits for it to exist. 

I'm wondering if we can do something similar with `smpmgr`. It waits for the device to be available. Once available, it flashes the binary. After the device is removed, `smpmgr` goes back to 'the top' and and waits again for the device to re-appear. At that time, it again flashes the binary... and on, and on..

The value is that
* It allows for an efficient way to flash many units. No need to even touch a keyboard! Just plug in a device, wait, unplug, and plug in the next one
* Allows for the fastest possible bootloader timeout (if its set that way), minimizing negative UX impact.

thoughts?

**Comment by JPHutchins** [comment]
I think this is doable now that the connect function has a timeout: https://github.com/intercreate/smpclient/blob/9895e9822c8bc6765ac3b95f210ab75d28491e58/smpclient/__init__.py#L36-L39

**Comment by JPHutchins** [comment]
Swinging back around on this, it's relevant that smpmgr could upgrade entire fleets of 100s of devices in parallel. This might be a good start - connect to all ports as they popup, including in parallel. I've actually already implemented and tested it once for a different project that was using smpclient.

Clean way might be to filter by VID:PID.
